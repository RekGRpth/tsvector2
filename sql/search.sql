-- without indexes
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'wr|qh';
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'wr&qh';
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'eq&yt';
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'eq|yt';
SELECT count(*) FROM test_tsvector2 WHERE a @@ '(eq&yt)|(wr&qh)';
SELECT count(*) FROM test_tsvector2 WHERE a @@ '(eq|yt)&(wr|qh)';
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'w:*|q:*';
SELECT count(*) FROM test_tsvector2 WHERE a @@ any ('{wr,qh}');
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'no_such_lexeme';
SELECT count(*) FROM test_tsvector2 WHERE a @@ '!no_such_lexeme';

create index wowidx on test_tsvector2 using gist (a);

SET enable_seqscan=OFF;
SET enable_indexscan=ON;
SET enable_bitmapscan=OFF;

explain (costs off) SELECT count(*) FROM test_tsvector2 WHERE a @@ 'wr|qh';

SELECT count(*) FROM test_tsvector2 WHERE a @@ 'wr|qh';
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'wr&qh';
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'eq&yt';
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'eq|yt';
SELECT count(*) FROM test_tsvector2 WHERE a @@ '(eq&yt)|(wr&qh)';
SELECT count(*) FROM test_tsvector2 WHERE a @@ '(eq|yt)&(wr|qh)';
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'w:*|q:*';
SELECT count(*) FROM test_tsvector2 WHERE a @@ any ('{wr,qh}');
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'no_such_lexeme';
SELECT count(*) FROM test_tsvector2 WHERE a @@ '!no_such_lexeme';

SET enable_indexscan=OFF;
SET enable_bitmapscan=ON;

explain (costs off) SELECT count(*) FROM test_tsvector2 WHERE a @@ 'wr|qh';

SELECT count(*) FROM test_tsvector2 WHERE a @@ 'wr|qh';
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'wr&qh';
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'eq&yt';
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'eq|yt';
SELECT count(*) FROM test_tsvector2 WHERE a @@ '(eq&yt)|(wr&qh)';
SELECT count(*) FROM test_tsvector2 WHERE a @@ '(eq|yt)&(wr|qh)';
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'w:*|q:*';
SELECT count(*) FROM test_tsvector2 WHERE a @@ any ('{wr,qh}');
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'no_such_lexeme';
SELECT count(*) FROM test_tsvector2 WHERE a @@ '!no_such_lexeme';

RESET enable_seqscan;
RESET enable_indexscan;
RESET enable_bitmapscan;

DROP INDEX wowidx;

CREATE INDEX wowidx ON test_tsvector2 USING gin (a);

SET enable_seqscan=OFF;
-- GIN only supports bitmapscan, so no need to test plain indexscan

explain (costs off) SELECT count(*) FROM test_tsvector2 WHERE a @@ 'wr|qh';

SELECT count(*) FROM test_tsvector2 WHERE a @@ 'wr|qh';
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'wr&qh';
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'eq&yt';
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'eq|yt';
SELECT count(*) FROM test_tsvector2 WHERE a @@ '(eq&yt)|(wr&qh)';
SELECT count(*) FROM test_tsvector2 WHERE a @@ '(eq|yt)&(wr|qh)';
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'w:*|q:*';
SELECT count(*) FROM test_tsvector2 WHERE a @@ any ('{wr,qh}');
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'no_such_lexeme';
SELECT count(*) FROM test_tsvector2 WHERE a @@ '!no_such_lexeme';

SELECT * FROM tsvector2_stat('SELECT a FROM test_tsvector2') ORDER BY ndoc DESC, nentry DESC, word LIMIT 10;
SELECT * FROM tsvector2_stat('SELECT a FROM test_tsvector2', 'AB') ORDER BY ndoc DESC, nentry DESC, word;

RESET enable_seqscan;
DROP EXTENSION tsvector2 CASCADE;
DROP TABLE test_tsvector2 CASCADE;
