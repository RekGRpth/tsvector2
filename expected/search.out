-- without indexes
SELECT count(*) FROM test_tsvector2 WHERE a @@ 'wr|qh';
 count 
-------
   158
(1 row)

SELECT count(*) FROM test_tsvector2 WHERE a @@ 'wr&qh';
 count 
-------
    17
(1 row)

SELECT count(*) FROM test_tsvector2 WHERE a @@ 'eq&yt';
 count 
-------
     6
(1 row)

SELECT count(*) FROM test_tsvector2 WHERE a @@ 'eq|yt';
 count 
-------
    98
(1 row)

SELECT count(*) FROM test_tsvector2 WHERE a @@ '(eq&yt)|(wr&qh)';
 count 
-------
    23
(1 row)

SELECT count(*) FROM test_tsvector2 WHERE a @@ '(eq|yt)&(wr|qh)';
 count 
-------
    39
(1 row)

SELECT count(*) FROM test_tsvector2 WHERE a @@ 'w:*|q:*';
 count 
-------
   494
(1 row)

SELECT count(*) FROM test_tsvector2 WHERE a @@ any ('{wr,qh}');
 count 
-------
   158
(1 row)

SELECT count(*) FROM test_tsvector2 WHERE a @@ 'no_such_lexeme';
 count 
-------
     0
(1 row)

SELECT count(*) FROM test_tsvector2 WHERE a @@ '!no_such_lexeme';
 count 
-------
   508
(1 row)
CREATE INDEX wowidx ON test_tsvector2 USING gin (a);
SET enable_seqscan=OFF;
-- GIN only supports bitmapscan, so no need to test plain indexscan
explain (costs off) SELECT count(*) FROM test_tsvector2 WHERE a @@ 'wr|qh';
                         QUERY PLAN                          
-------------------------------------------------------------
 Aggregate
   ->  Bitmap Heap Scan on test_tsvector2
         Recheck Cond: (a @@ '''wr'' | ''qh'''::tsquery)
         ->  Bitmap Index Scan on wowidx
               Index Cond: (a @@ '''wr'' | ''qh'''::tsquery)
(5 rows)

SELECT count(*) FROM test_tsvector2 WHERE a @@ 'wr|qh';
 count 
-------
   158
(1 row)

SELECT count(*) FROM test_tsvector2 WHERE a @@ 'wr&qh';
 count 
-------
    17
(1 row)

SELECT count(*) FROM test_tsvector2 WHERE a @@ 'eq&yt';
 count 
-------
     6
(1 row)

SELECT count(*) FROM test_tsvector2 WHERE a @@ 'eq|yt';
 count 
-------
    98
(1 row)

SELECT count(*) FROM test_tsvector2 WHERE a @@ '(eq&yt)|(wr&qh)';
 count 
-------
    23
(1 row)

SELECT count(*) FROM test_tsvector2 WHERE a @@ '(eq|yt)&(wr|qh)';
 count 
-------
    39
(1 row)

SELECT count(*) FROM test_tsvector2 WHERE a @@ 'w:*|q:*';
 count 
-------
   494
(1 row)

SELECT count(*) FROM test_tsvector2 WHERE a @@ any ('{wr,qh}');
 count 
-------
   158
(1 row)

SELECT count(*) FROM test_tsvector2 WHERE a @@ 'no_such_lexeme';
 count 
-------
     0
(1 row)

SELECT count(*) FROM test_tsvector2 WHERE a @@ '!no_such_lexeme';
 count 
-------
   508
(1 row)

SELECT * FROM tsvector2_stat('SELECT a FROM test_tsvector2') ORDER BY ndoc DESC, nentry DESC, word LIMIT 10;
 word | ndoc | nentry 
------+------+--------
 qq   |  108 |    108
 qt   |  102 |    102
 qe   |  100 |    100
 qh   |   98 |     98
 qw   |   98 |     98
 qa   |   97 |     97
 ql   |   94 |     94
 qs   |   94 |     94
 qi   |   92 |     92
 qr   |   92 |     92
(10 rows)

SELECT * FROM tsvector2_stat('SELECT a FROM test_tsvector2', 'AB') ORDER BY ndoc DESC, nentry DESC, word;
 word | ndoc | nentry 
------+------+--------
(0 rows)

RESET enable_seqscan;
DROP EXTENSION tsvector2 CASCADE;
NOTICE:  drop cascades to column a of table test_tsvector2
DROP TABLE test_tsvector2 CASCADE;
